src/backend/utils/misc/queryenvironment.c

```diff
@@ -435,25 +435,27 @@ bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List
 				* Search through the entire ENR relation list for everything
 				* that has a relation (non-recursive) to this object.
 				* If indexId is DependDependerIndexId, we try to mimic
-				* SELECT * FROM pg_depend WHERE classid=v1 AND objid=v2
+				* SELECT * FROM pg_depend WHERE classid=v1 AND objid=v2 (AND objsubid = v3 if applicable)
 				* Otherwise if it is DependReferenceIndexId we try to mimic
-				* SELECT * FROM pg_depend WHERE refclassid=v1 AND refobjid=v2
+				* SELECT * FROM pg_depend WHERE refclassid=v1 AND refobjid=v2 (AND refobjsubid = v3 if applicable)
 				* So we cannot return right away if there is a match.
 				*/
 				ListCell   *lc;
 				foreach(lc, enr->md.cattups[ENR_CATTUP_DEPEND]) {
 					Form_pg_depend tup = (Form_pg_depend) GETSTRUCT((HeapTuple) lfirst(lc));
 					if (indexId == DependDependerIndexId &&
 						tup->classid == (Oid)v1 &&
-						tup->objid == (Oid)v2)
+						tup->objid == (Oid)v2 &&
+						(nkeys == 2 || tup->objsubid == (int32)v3))
```

Comment:
is it safe to just check if tup->objsubid == v3? since v3 if not set is 0, and tup->objsubid will be 0 for non-column lookups

Reply:
That doesn't work - if the caller of this function only wants 2 keys, then they want all pg_depend entries that match the objid, no matter what the subid is - so we need to completely ignore the subid in that case.